lib/simplecov/defaults.rb

```diff
@@ -79,15 +80,18 @@
         $stderr.printf("File (%s) is only (%.2f%%) covered. This is below the expected minimum coverage per file of (%.2f%%).\n", SimpleCov.result.least_covered_file, covered_percentages.min, SimpleCov.minimum_coverage_by_file)
         @exit_status = SimpleCov::ExitCodes::MINIMUM_COVERAGE
       elsif (last_run = SimpleCov::LastRun.read) # rubocop:disable Metrics/BlockNesting
-        diff = last_run["result"]["covered_percent"] - covered_percent
-        if diff > SimpleCov.maximum_coverage_drop # rubocop:disable Metrics/BlockNesting
-          $stderr.printf("Coverage has dropped by %.2f%% since the last time (maximum allowed: %.2f%%).\n", diff, SimpleCov.maximum_coverage_drop)
+        coverage_diff = last_run["result"]["covered_percent"] - covered_percent
+        if coverage_diff > SimpleCov.maximum_coverage_drop # rubocop:disable Metrics/BlockNesting
+          $stderr.printf("Coverage has dropped by %.2f%% since the last time (maximum allowed: %.2f%%).\n", coverage_diff, SimpleCov.maximum_coverage_drop)
           @exit_status = SimpleCov::ExitCodes::MAXIMUM_COVERAGE_DROP
         end
       end
     end
 
-    SimpleCov::LastRun.write(:result => {:covered_percent => covered_percent})
+    # Don't overwrite last_run file if refuse_coverage_drop option is enabled and the coverage has dropped
+    unless SimpleCov.refuse_coverage_drop && coverage_diff > 0
```

Comment:
If I understand this code correctly, then the call to `SimpleCov.refuse_coverage_drop` will refuse a coverage drop. If I'm right we shouldn't do this and if the tests really don't catch this I should add a test :D